### ISSUE TITLE
pip-installed numpy wheel not importable within conda environment

### ISSUE THREAD

**Issue by RobertLucian** [conda/conda#6423]
## Numpy failing in conda's root environment

1. Starting with Jessie Lite as a base image on a Raspberry Pi, I install miniconda3 and then I proceed to install pip with `conda install pip`. This is all done on the `root` environment of conda.

2. Next, I install the numpy wheel with the following command:
```
pip install numpy --extra-index-url https://www.piwheels.hostedpi.com/simple
```

3. Then, I try importing the numpy module inside the python (v3.4.3) interpreter and I get the following error:
```

Python 3.4.3 |Continuum Analytics, Inc.| (default, Aug 21 2015, 00:53:08)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.2.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numpy
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-d4cdadb62aa7> in <module>()
----> 1 import numpy

/opt/miniconda3/lib/python3.4/site-packages/numpy/__init__.py in <module>()
    164     from . import fft
    165     from . import polynomial
--> 166     from . import random
    167     from . import ctypeslib
    168     from . import ma

/opt/miniconda3/lib/python3.4/site-packages/numpy/random/__init__.py in <module>()
     97 with warnings.catch_warnings():
     98     warnings.filterwarnings("ignore", message="numpy.ndarray size changed")
---> 99     from .mtrand import *
    100
    101 # Some aliases:

ImportError: /opt/miniconda3/lib/python3.4/site-packages/numpy/random/mtrand.cpython-34m.so: undefined symbol: PyFPE_jbuf
```
I've done this multiple of times, so I'm fairly sure this is repeatable.

## Numpy imported within system-wide space

Next, I did something else, and that was to reflash the sd card with a fresh Jessie Lite again (we wouldn't want to have the environment contaminated at all) and install numpy system-wide with conda not installed.
Basically, I did the following things:
```
$ sudo apt-get install python3 python3-pip
$ sudo pip install numpy --extra-index-url https://www.piwheels.hostedpi.com/simple
```
Afterward, I launched the python interpreter and imported the numpy module, and it worked.

## Bottom Line

Bottom line is that on a Raspberry Pi, while in conda's environment, a pip-installed numpy module can't be imported whilst in the system-wide environment it works fine. The same wheel has been used for installing in both scenarios.

Now, you might ask yourself why haven't I used the `conda install numpy` command?
Well, first of all, keep in mind I'm developing a system for installing packages in offline mode (non interactive mode).

The issue is a lot of python packages can't be installed from continuum's repositories on a Raspberry Pi, because they are yet to be available. Because of this, I need to fill the void by installing packages with pip, but doing this creates another 2 problems:
1. Because I'm installing packages with pip and conda, there are moments when I get the infamous `Cannot remove entries from nonexistent file ***/site-pack
ages/easy-install.pth`  error, which can by the way, be avoided by adding the `--ignore-installed` option. Even with that, it makes me nervous thinking that errors like these can appear. And when updates are being done without a human's supervision, things can get ugly - so I can't trust that everything will go smoothly all the time.
2. A mechanism for installing/updating packages with conda and pip together is going to be inherently more complex than having just one tool. And since I'm only interested in installing python packages, pip seems to be the best choice, because there are a large number of packages. The only problem with the packages on PyPi is that for many of them, I don't get the wheels, but instead, the source code, which needs to be build. As an alternative to that, there is https://github.com/bennuttall/piwheels, which has pre-compiled wheels.

So as you can see, due to the lack of conda packages for python for the `armv7` architecture, I'm going to be stuck with pip and piwheels. I'm still going to use conda as an environment manager. To me, the conda + pip combo looks better than virtualenv + pip.

## TL;DR

Pip-installed numpy package can't be imported in conda's environment while being on a Raspberry Pi. Is there anyone else who's encountering this issue? Does anyone know what the problem is?

## Update

When pip installing from the source, it works (takes roughly 30 minutes on a Pi 3).

**Comment by kalefranz** [comment]
I don't think there's anything I can change in conda's own code to fix this for you.  You're pip-installing something that has dependencies outside of python, which is exactly the problem conda was created to solve.  Actually, if there's any package conda was created specifically to help solve install problems for, it's `numpy`.

You might find https://github.com/jjhelmus/berryconda helpful.
